src/tuf.rs

```diff
@@ -333,3 +452,42 @@ impl TargetCustom {
         TargetCustom { name, version, uri, hardware_ids, created_at, updated_at }
     }
 }
+
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub struct PublicKey {
+    keytype: crypto::KeyType,
+    keyval: PublicKeyValue,
+}
+
+impl PublicKey {
+    pub fn from_file(typ: crypto::KeyType, path: &str) -> Result<Self> {
+        let mut file = File::open(path)?;
+        let mut public = String::new();
+        file.read_to_string(&mut public)?;
+        Ok(PublicKey { keytype: typ, keyval: PublicKeyValue { public } })
+    }
+
+    pub fn from_pubkey(typ: crypto::KeyType, public: &crypto::PubKeyValue) -> Result<Self> {
+        Ok(PublicKey {
+            keytype: typ,
+            keyval: PublicKeyValue {
+                public: match typ {
+                    crypto::KeyType::Ed25519 => HEXLOWER.encode(&*public),
```

Comment:
We don't use hexlower anymore. That nightmare of DER parsing/writing is to make the rust program place nice with the format expected by Bouncy Castle on the backend.